src/babashka/nrepl/impl/server.clj

```diff
@@ -46,7 +52,14 @@
                 (set! *1 value)
                 (utils/send o (utils/response-for msg
                                                   {"ns" (vars/current-ns-name)
-                                                   "value" (pr-str value)}) opts)
+                                                   "value" (if nrepl-pprint
+                                                             (if-let [pprint-fn (get nrepl-pprint pretty-print-fns-map)]
+                                                               (pprint-fn value)
```

Comment:
@kolharsam Note that `clojure.pprint/pprint` prints to `*out*`. To return a string we'll need to use `with-out-str` here.